### PR TITLE
Mobile Release v1.70.3

### PR DESCRIPTION
**Pull request by mchowning**
## Description
Release 1.70.3 of the react-native-editor and Gutenberg-Mobile.

For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4565

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] I've tested my changes with keyboard and screen readers. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal). 
- [ ] I've updated related schemas if appropriate.

### CODE REVIEW

**Review by SiobhyB — APPROVED** [review]
Approved via https://github.com/wordpress-mobile/gutenberg-mobile/pull/4565#pullrequestreview-879045791

**Review by fluiddot — APPROVED** [review]

File: packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -764,7 +764,7 @@
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
 				COPY_PHASE_STRIP = NO;
 				ENABLE_STRICT_OBJC_MSGSEND = YES;
-				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "arm64 ";
+				"EXCLUDED_ARCHS[sdk=iphonesimulator*]" = "";
```
Comment (fluiddot) [inline packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj:767]: Looks like these changes were introduced in `1.70.3` but I don't recall any modification in the fix ([reference](https://github.com/WordPress/gutenberg/pull/38670)) related to this. @mchowning I'm wondering if you have any clue about where this change came from, thanks 🙇 .

Comment (mchowning) [reply packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj:767]: I don't. 🤔  

It looks like @geriux made the opposite change [for the 1.70.2 release](https://github.com/WordPress/gutenberg/commit/323791acd67b1cdabaa67d48d8d4e1a077b2c23a). Perhaps he has some insight.

Comment (mchowning) [reply packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj:767]: Before that, it actually looks like @fluiddot  😄  was the last person to add to add the "arm64" exclusion: https://github.com/WordPress/gutenberg/commit/a91013931d893f8778c9986713060111e1115745

Comment (fluiddot) [reply packages/react-native-editor/ios/GutenbergDemo.xcodeproj/project.pbxproj:767]: Oh, good catch, I didn't recall the change 😅. If the change was introduced in https://github.com/WordPress/gutenberg/commit/323791acd67b1cdabaa67d48d8d4e1a077b2c23a for the `1.70.2` release, I'm wondering why it's not in `trunk` 🤔. In any case, I think the change is valid as it's not expected that we exclude the `arm64` architecture for the simulator hence, I'll go ahead and merge the PR once the CI checks pass.

**Pull request merged**